### DISCUSSION TITLE
ws2812b light wont work, even on root.

### DISCUSSION THREAD

**Discussion by KnoppKnopp** [awawa-dev/HyperHDR#152] [Q&A]
Can anyone help me figure out the issue with my LEDs please.

I originally had the sk6812 leds but they caused way to many issues and so i bought the ws2812b strip instead

I cannot for the life of me get them to light up despite the entire setup appearing to work perfectly.

This is my full log on startup after changing to root.
https://pastebin.com/iYAiDbHx

I'm at a loss. 

<img width="1016" alt="Screenshot 2021-11-14 at 12 42 58" src="https://user-images.githubusercontent.com/92688938/141684183-3785fba9-8026-47a0-b819-cce7ed68fb9e.png">
<img width="1006" alt="Screenshot 2021-11-14 at 12 42 34" src="https://user-images.githubusercontent.com/92688938/141684185-5dcf4006-f26b-4964-82ff-d1992d41fd58.png">
<img width="1193" alt="Screenshot 2021-11-14 at 12 42 20" src="https://user-images.githubusercontent.com/92688938/141684186-24b5305e-ff5c-4ae3-84aa-3d6f11df011c.png">

**Comment by awawa-dev** [comment]
Log looks fine and configuration looks similar to reference ([link](https://github.com/awawa-dev/HyperHDR/discussions/111)), changing sk6812 to ws2812b is a serious downgrade and the problem (probably electrical nature) remain the same. If you don't have a voltage level shifter(for example SN74AHCT125N) then I suggest to install it as first step, because it's cheapest and easiest solution that may help. Anyway it's almost always a lottery that depends on particural setup build (for example cable lenghts and the voltage provided by the power supply). Without a multimeter and ESP device with working WLED there is no much to diagnose further.
![obraz](https://user-images.githubusercontent.com/69086569/141684969-20d04a06-415f-4b43-bd22-aa6308f54c25.png)

**Reply by KnoppKnopp** [reply]
Thanks for your response.
I understand, I could not get the SK6812 to work either. I could get them to light up and follow the screen colours but there was always a constant ficker which I could not resolve. I originally thought it was a ground issue but it turned out thatr wasnt the case. I used to logic level converter for the sk6812 setup with no luck, same with this setup. Its driving me mad because they were working a month ago. I updated from v16 to v17 and nothing works at all. Even if I backdate.

**Reply by awawa-dev** [reply]
One of my Sk6812 setup broke and the flickering effect was similar to yours: I needed to replace 3 leds (the one from the flickering was staring and both its neighbors to be sure). Which level shifter you were using: that one [link](https://cdn.sparkfun.com/r/500-500/assets/parts/8/5/2/2/12009-06.jpg)?

**Reply by KnoppKnopp** [reply]
The flickering applies to the entire strip in my case. It would work perfect except it would flicker every 10 seconds or so. This is why I thought it to be a dodgy ground issue. I tried every combination of solutions and none worked. 

This is the level converter I tried. I assume they do the same job? [link](https://www.amazon.co.uk/gp/product/B07Y86JBZY/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&psc=1)

**Reply by awawa-dev** [reply]
Well, it can convert voltage level that matter and looks similar to sparkfun model its probably too slow for our purpose and high speed data transfer: so still we are in the 'lottery' territory with may or may not work. If the whole LED strip was flickering only from time to time then it's probably different case than mine. Anyway you should measure the voltage that is provided to the LED strip, if right get SN74AHCT125N and at least ESP8266 wemos d1 mini to test it with WLED as a reference system.

**Reply by KnoppKnopp** [reply]
Thanks for the advice. I will give that ago. 

Just for context, this was the issue I have with the sk6812 strip
[link](https://imgur.com/a/ZTuituC)

**Reply by awawa-dev** [reply]
Well, that was similar to one of my old WS2801 setup (without a level shifter, but I don't know the cause...power supply with high voltage spikes also could play its role) that I mentioned here [link](https://hyperhdr.blogspot.com/2020/12/my-build-log-using-sk6812-rgbw-led.html#chapter3) but it wasn't so bad.

With new setup try to not exceed 30-40cm data line from Rpi (or ESp8266 with WLED) to the beginning of the LED strip to reduce EMI interference.
